crates/bevy_text/src/font_atlas.rs

```diff
@@ -118,3 +118,139 @@ impl core::fmt::Debug for FontAtlas {
             .finish()
     }
 }
+
+/// Adds the given subpixel-offset glyph to the given font atlases
+pub fn add_glyph_to_atlas(
+    font_atlases: &mut Vec<FontAtlas>,
+    texture_atlases: &mut Assets<TextureAtlasLayout>,
+    textures: &mut Assets<Image>,
+    font_system: &mut cosmic_text::FontSystem,
+    swash_cache: &mut cosmic_text::SwashCache,
+    layout_glyph: &cosmic_text::LayoutGlyph,
+    font_smoothing: FontSmoothing,
+) -> Result<GlyphAtlasInfo, TextError> {
+    let physical_glyph = layout_glyph.physical((0., 0.), 1.0);
+
+    let (glyph_texture, offset) =
+        get_outlined_glyph_texture(font_system, swash_cache, &physical_glyph, font_smoothing)?;
+    let mut add_char_to_font_atlas = |atlas: &mut FontAtlas| -> Result<(), TextError> {
+        atlas.add_glyph(
+            textures,
+            texture_atlases,
+            physical_glyph.cache_key,
+            &glyph_texture,
+            offset,
+        )
+    };
+    if !font_atlases
+        .iter_mut()
+        .any(|atlas| add_char_to_font_atlas(atlas).is_ok())
+    {
+        // Find the largest dimension of the glyph, either its width or its height
+        let glyph_max_size: u32 = glyph_texture
+            .texture_descriptor
+            .size
+            .height
+            .max(glyph_texture.width());
+        // Pick the higher of 512 or the smallest power of 2 greater than glyph_max_size
+        let containing = (1u32 << (32 - glyph_max_size.leading_zeros())).max(512);
+        font_atlases.push(FontAtlas::new(
+            textures,
+            texture_atlases,
+            UVec2::splat(containing),
+            font_smoothing,
+        ));
+
+        font_atlases.last_mut().unwrap().add_glyph(
+            textures,
+            texture_atlases,
+            physical_glyph.cache_key,
+            &glyph_texture,
+            offset,
+        )?;
+    }
+
+    Ok(get_glyph_atlas_info(font_atlases, physical_glyph.cache_key).unwrap())
+}
+
+/// Get the texture of the glyph as a rendered image, and its offset
+pub fn get_outlined_glyph_texture(
```

Comment:
Maybe we can change the visibility to private. I don't think anything is depending on this function externally.
```suggestion
fn get_outlined_glyph_texture(
```